ext/socket/lib/socket.rb

```diff
@@ -593,6 +594,74 @@ def accept_nonblock(exception: true)
     __accept_nonblock(exception)
   end
 
+  def self.start_getaddrinfo_v6(host, port, queue, resolv_timeout)
+    return Thread.new do
+      ai_list = Addrinfo.getaddrinfo(host, port, :PF_INET6, :STREAM, timeout: resolv_timeout)
+      ai_list.each {|ai| queue.push(ai) }
+    rescue SocketError => ex
+      case ex.message
+      when "getaddrinfo: Address family for hostname not supported" # when IPv6 is not supported
```

Comment:
I think error messages are not reliable.
It seems that SocketError needs to be extended.
(There are several choices to extend SocketError such as
adding accessor to error code, creating bunch of subclasses like Errno::*, etc.
I'm not sure what matz like.)

Reply:
We can't retrieve errno or other reliable information from SocketError.
I will open another ticket to extend SocketError.

Reply:
I opened a PR about this: #4247